rocAL/rocAL/source/pipeline/tensor.cpp

```diff
@@ -320,17 +374,39 @@ unsigned rocalTensor::copy_data(hipStream_t stream, void *host_memory, bool sync
 unsigned rocalTensor::copy_data(void *user_buffer) {
     if (_mem_handle == nullptr) return 0;
 
-#if ENABLE_HIP
-    if (_info._mem_type == RocalMemType::HIP) {
-        // copy from device to host
-        hipError_t status;
-        if ((status = hipMemcpyDtoH((void *)user_buffer, _mem_handle, _info.data_size())))
-            THROW("copy_data::hipMemcpyDtoH failed: " + TOSTR(status))
-    } else
-#endif
-    {
-        // copy from host to host
-        memcpy(user_buffer, _mem_handle, _info.data_size());
+    #if ENABLE_HIP
+        if (_info._mem_type == RocalMemType::HIP) {
+            // copy from device to device
+            hipError_t status;
+            if ((status = hipMemcpyDtoD((void *)user_buffer, _mem_handle, _info.data_size())))
+                THROW("copy_data::hipMemcpyDtoH failed: " + TOSTR(status))
+        } else if (_info._mem_type == RocalMemType::HOST) {
+            // copy from host to device
+            hipError_t status;
+            if ((status = hipMemcpyHtoD((void *)user_buffer, _mem_handle, _info.data_size())))
+                THROW("copy_data::hipMemcpyHtoD failed: " + TOSTR(status))
+        }
+    #else
+        if (_info._mem_type == RocalMemType::HOST)
+            memcpy(user_buffer, _mem_handle, _info.data_size());
+    #endif
+
+    return 0;
+}
+
+unsigned rocalTensor::copy_data(void *user_buffer, uint max_y1, uint max_x1) {
+    // if (_mem_handle == nullptr) return 0;
+    //TODO : Handle this case for HIP buffer
+    auto src_stride = (_info.max_shape().at(0) * _info.max_shape().at(1) * _info.data_type_size());
+    auto dst_stride = (max_y1 * max_x1 * _info.data_type_size());
+    for (uint i = 0; i < _info._batch_size; i++) {
+        auto temp_src_ptr = static_cast<unsigned char *>(_mem_handle) + i * src_stride;
+        auto temp_dst_ptr = static_cast<unsigned char *>(user_buffer) + i * dst_stride;
+        for (uint height = 0; height < max_y1; height++) {
+            memcpy(temp_dst_ptr, temp_src_ptr, max_x1 * _info.data_type_size());
+            temp_src_ptr += _info.max_shape().at(0) * _info.data_type_size();
+            temp_dst_ptr += max_x1 * _info.data_type_size();
```

Comment:
Let's store the _info.data_type_size(), _info.max_shape().at(0) also in another variables before loop instead of querying every time

Reply:
Done